packages/slate-react/src/components/string.tsx

```diff
@@ -58,8 +58,18 @@ const String = (props: {
 
 const TextString = (props: { text: string; isTrailing?: boolean }) => {
   const { text, isTrailing = false } = props
+  const editor: any = useEditor()
+
+  const ref: React.RefObject<HTMLSpanElement> = useRef() as any
+
+  useEffect(() => {
+    if (ref.current && ref.current.innerText !== text) {
+      ref.current.innerText = text
+    }
+  })
```

Comment:
What's the goal of this `useEffect` statement here? Can you add a clear comment about with `COMPAT: ...` explaining this for future readers.

Reply:
Sure, I will commit it soon

Reply:
Yeah, I want to find a more *React* way to force the `String` node to be updated at first.But then I found the main reason of this issue is because **React can't detect this kind change of dom**. If someone find a more *React* way to solve this issue, that will be better.

The reason why `useLayoutEffect` just fix this problem is **selection is reset at `Editable` component** which is the root of all `String` components. The direction of React run effects is from leaf to root, so the text reset will run first. And `useEffect` is async which run on nextTick, so it won't work with the selection reset in `layoutEffect`.

`reconciliation` just do something with virtual dom, it never touch the real dom